src/plugins/intel_gna/src/frontend/scale_factor_calc.cpp

```diff
@@ -1315,18 +1316,18 @@ bool ScaleFactorCalculator::ScaleFactorPerLayerGemm(InferenceEngine::GemmLayer*
     // Adjust weights scale factor if output values exceed int32 maximum value
     auto weightsReducer = calculateWeightsReducerFromDstStats(quantData->_dst_quant);
     if (LayerInfo(in0).isConst()) {
-        if (!common::fp32eq(weightsReducer, 1.0f)) {
+        if (!are_fp_eq(static_cast<float>(weightsReducer), 1.0f)) {
```

Comment:
Why casting if weightsReducer is double?

Reply:
Some of tests functional tests being part of smoke_base/MatMulOverloadCorrectio are failing if precision is to high.